src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -86,7 +86,7 @@ protected function formatValue(mixed $value, int $format = 0): string
         }
 
         if ($value instanceof \UnitEnum) {
-            return $value->name;
+            $value = $value instanceof \BackedEnum ? $value->value : $value->name;
```

Comment:
I personally don't think this is a good idea. 

Enum case names are helpful  for developers, scalar equivalent less so.  

Moreover, this will have an impact on other constraints, right ?

Reply:
To be honest, I don't need this enum functionality in choice constraint (just had an idea & started coding), but this line of code should be fixed upstream. In the `BackedEnumNormalizer` enums are currently (de)normalized from/to `->value`, here it's `->name`. Using some `{{ value }}` parameter in the messages block can confuse some, because input value is not same as displayed.

Btw this should not break anything, may third-party tests depending on this functionality. Haven't seen much enum usages in constraints and tests worked smooth for the other part of code.

PS: Is there any reason why there's no `UnitEnumNormalizer`, just `BackedEnumNormalizer`? If it makes sense to implement (for me it does), I'll create a new MR for that :-)